### ISSUE TITLE
alignment is bugged

### ISSUE THREAD

**Issue by DanySK** [Protelis/Protelis#83]
Calling this function

``` scala
public def planB() {
    env.put("help", gradCastOnMap(hurtsInSight() > 3, self.getDevicePosition()));
    mux (hurtsInSight() > 0) {
        follow(gradCastOnMap(env.has("hurt"), self.getDevicePosition()))
    } else {
        0
    }
}
```

makes protelis throw an error (field defined twice at the same code point). The following:

``` scala
public def planB() {
    env.put("help", gradCastOnMap(hurtsInSight() > 3, self.getDevicePosition()));
    1;
    mux (hurtsInSight() > 0) {
        follow(gradCastOnMap(env.has("hurt"), self.getDevicePosition()))
    } else {
        0
    }
}
```

has no issues instead.

I believe we could fix this by assigning each operation a specific opcode to be pushed onto the current "codepath" structure, rather than a positional argument as it is now.

**Comment by jakebeal** [comment]
I don't understand what's causing the bug currently: which two elements are being defined as identical, and why?

**Comment by DanySK** [comment]
The issue is likely caused by bytecodes of Protelis instructions overlapping: the second entry in an All block, which emulates multi-statement expressions, has the same byte code of the mux condition.
We probably need to tackle this in two ways:
1. Assigning a bytecode to any interpreter entity
2. Assigning a bytecode to any interpreter sub-entity (e.g. if condition, if then clause, etc)
3. Thinking of a way to encode bytecodes for the all block in such a way that they do not overlap with others

**Comment by DanySK** [comment]
Fixed in #201
